apps/game-service/src/etc/npc/npc-spawner.ts

```diff
@@ -11,31 +11,73 @@ import { NpcInstance } from "@mp/game-shared";
 import { cardinalDirections, clamp, Vector } from "@mp/math";
 import type { VectorGraphNode } from "@mp/path-finding";
 import type { Rng, Tile } from "@mp/std";
-import { assert, createShortId } from "@mp/std";
+import { assert, createShortId, promiseFromResult, withBackoffRetries } from "@mp/std";
 import type { TickEventHandler } from "@mp/time";
 import { TimeSpan } from "@mp/time";
 import { deriveNpcSpawnsFromArea } from "./derive-npc-spawns-from-areas";
+import type { DbRepository } from "@mp/db";
+import type { Logger } from "@mp/logger";
 
 interface NpcSpawnOption {
   spawn: NpcSpawn;
   npc: NpcDefinition;
 }
 
 export class NpcSpawner {
-  public readonly options: ReadonlyArray<NpcSpawnOption>;
+  private options: ReadonlyArray<NpcSpawnOption> = [];
 
   constructor(
     private readonly area: AreaResource,
     private readonly models: ActorModelLookup,
-    optionsFromDB: Array<NpcSpawnOption>,
     private readonly rng: Rng,
+    private readonly db: DbRepository,
+    private readonly logger: Logger,
   ) {
-    const optionsFromTiled = deriveNpcSpawnsFromArea(
-      this.area,
-      optionsFromDB.map(({ npc }) => npc),
-    );
+    // Start lazy loading spawn options with infinite retry
+    void this.lazyLoadSpawnOptions();
+  }
 
-    this.options = [...optionsFromDB, ...optionsFromTiled];
+  private async lazyLoadSpawnOptions() {
+    let attempt = 0;
+
+    while (this.options.length === 0) {
+      try {
+        attempt++;
+        this.logger.info({ attempt }, `Loading NPC spawn options...`);
+
+        // Use withBackoffRetries with capped delay
+        // oxlint-disable-next-line no-await-in-loop
+        const optionsFromDB = await withBackoffRetries(
+          () => promiseFromResult(this.db.selectAllSpawnAndNpcPairs(this.area.id)),
+          {
+            maxRetries: 10, // Try 10 times before letting outer loop retry
```

Comment:
implement support for `maxRetries: "infinite"` in `withBackoffRetries` and use that in all usages in this PR.